tests/Unit/Options/Test_Factory.cpp

```diff
@@ -86,18 +88,47 @@ struct Map {
   static constexpr OptionString help = {"halp"};
 };
 
+template <bool Valid>
+struct Metavars {
+  static constexpr bool valid = Valid;
+};
+
+struct TestWithMetavars : OptionTest {
+  struct Arg {
+    using type = std::string;
+    static constexpr OptionString help = {"halp"};
+  };
+  using options = tmpl::list<Arg>;
+  static constexpr OptionString help = {""};
+
+  TestWithMetavars() = default;
+  template <typename Metavariables>
+  explicit TestWithMetavars(std::string arg, const OptionContext& /*unused*/,
+                            Metavariables /*meta*/)
+      : arg_(std::move(arg)), valid_(Metavariables::valid) {}
+
+  std::string name() const override {
+    return "TestWithArg(" + arg_ + ")" +
+           (valid_ ? std::string{"yes"} : std::string{"no"});
+  }
+
+ private:
+  std::string arg_;
+  bool valid_{false};
+};
+
 void test_factory() {
   Options<tmpl::list<OptionType>> opts("");
   opts.parse("OptionType: Test2");
-  CHECK(opts.get<OptionType>()->name() == "Test2");
+  CHECK(opts.get<OptionType, Metavars<true>>()->name() == "Test2");
```

Comment:
It's necessary because the `TestWithMetavars` derived class is in the `creatable_classes` list on line 32. No, I don't think it should have a default template parameter because then I think the error message would be more confusing when no metavars is passed. This way it forces people to think about it.

Reply:
I'll add comments in the test to clarify why we need to pass the metavars